diff -urN minidlna-1.1.4.org/minidlna.conf minidlna-1.1.4/minidlna.conf
--- minidlna-1.1.4.org/minidlna.conf	2014-08-26 23:09:22.000000000 +0200
+++ minidlna-1.1.4/minidlna.conf	2014-10-13 21:49:08.792041371 +0200
@@ -15,7 +15,7 @@
 #   + "V" for video  (eg. media_dir=V,/home/jmaggard/Videos)
 #   + "P" for images (eg. media_dir=P,/home/jmaggard/Pictures)
 #   + "PV" for pictures and video (eg. media_dir=PV,/home/jmaggard/digital_camera)
-media_dir=/opt
+#media_dir=/opt
 
 # set this to merge all media_dir base contents into the root container
 # note: the default is no
@@ -25,10 +25,10 @@
 #friendly_name=My DLNA Server
 
 # set this if you would like to specify the directory where you want MiniDLNA to store its database and album art cache
-#db_dir=/var/cache/minidlna
+db_dir=/var/cache/minidlna
 
 # set this if you would like to specify the directory where you want MiniDLNA to store its log file
-#log_dir=/var/log
+log_dir=/var/log/minidlna
 
 # set this to change the verbosity of the information that is logged
 # each section can use a different level: off, fatal, error, warn, info, or debug
